### DISCUSSION TITLE
Very slow lookup when hovering over a symbol

### DISCUSSION THREAD

**Discussion by kenroser** [microsoft/vscode-cpptools#8283] [Q&A]
On a Intel Macbook Pro, I believe I have vscode setup properly and am using compile_commands.json.  The .cpp files don't have any squiggly lines and all the headers are found.

My problem is when I hover over something to see what it is, "Loading..." appears and it takes about 40-60 seconds before the result appears.  Similarly if I hit F12, there's a very long delay.  

Using Process Explorer, I see in both cases a cpptools-srv process starts up with 100% CPU usage and stays running for the same waiting period I have to see what a symbol is.

VSCode and the C++ extension are all up-to-date as of today.

This problem is not seen with VSCode on Linux with the same software in the workspace.

I'm hoping someone can provide some guidance on resolving this issue.

**Comment by kenroser** [comment]
Following the directions at https://github.com/microsoft/vscode-cpptools/wiki/Troubleshooting-Performance-Issues, attached is the Instruments output for cpptools-srv when the problem exists.
![image](https://user-images.githubusercontent.com/18311007/137505158-a7c31534-43b8-45ca-b520-400399b7a452.png)

**Comment by kenroser** [comment]
If I turn on the C/C++ debugger, when hovering, an IntelliSense process crash is detected:

```textDocument/hover: file:///Users/roser/code/file.cpp (id: 85)
terminating child process: 18651
IntelliSense process crash detected.
cpptools/finishUpdateSquiggles
Update IntelliSense time (sec): 22.855
Checking for syntax errors: file:///Users/roser/code/file.cpp
Update IntelliSense time (sec): 0```

**Reply by sean-mcmanus** [reply]
What version of the C/C++ extension are you using? Are you able to get a crash call stack or are you saying attaching the debugger causes it to crash? I don't see anything unusual with the performance logs, but I'll check with others on our team...

**Reply by kenroser** [reply]
I was using 1.7.0 and also tried the previous one.  
I don't have a call stack and haven't tried attaching a debugger.
Don't you think it's unusual in my comment that it says "IntelliSense process crash detected."

I'm willing to do whatever you ask to help debug this.  Just let me know what to try.
